Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,183 @@
+                            SoundSystem.Play(Filter.Entities(uid), "/Audio/Effects/guardian_inject.ogg", uid);
+                            _popupSystem.PopupCoordinates(Loc.GetString("guardian-created"), hostcomp.Owner.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                            //Exhaust the activator
+                            component.Used = true;
+                        }                   
+                    }
+                    else
+                    {
+                        _popupSystem.PopupCoordinates(Loc.GetString("guardian-activator-empty-invalid-creaton"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                    }
+                }
+                else
+                {
+                    _popupSystem.PopupCoordinates(Loc.GetString("guardian-no-actions-invalid-creation"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                }
+            }
+            else
+            {
+                _popupSystem.PopupCoordinates(Loc.GetString("guardian-already-present-invalid-creation"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+            }
+        }
+
+        /// <summary>
+        /// Called every time the host moves, to make sure the distance between the host and the guardian isn't too far
+        /// </summary>
+        private void OnGuardianHostMove(EntityUid uid, GuardianHostComponent component, ref MoveEvent args)
+        {
+            if (EntityManager.GetEntity(component.Hostedguardian).TryGetComponent<GuardianComponent>(out GuardianComponent? guardcomp))
+            {
+                if (guardcomp.guardianLoose == true)
+                {
+                    //Compares the distance to allowed distance, otherwise forces a recall action from the host
+                    if (!guardcomp.Owner.Transform.Coordinates.InRange(EntityManager, component.Owner.Transform.Coordinates, guardcomp.DistanceAllowed))
+                    {
+                        OnGuardianManifestAction(guardcomp.OwnerUid, uid);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Called every time the guardian moves: makes sure it's not out of it's allowed distance
+        /// </summary>
+        private void OnGuardianMove(EntityUid uid, GuardianComponent component, ref MoveEvent args)
+        {
+            if (component.guardianLoose == true)
+            {
+                var guardian = EntityManager.GetEntity(uid);
+                var host = EntityManager.GetEntity(component.Host);
+                //Compares the distance to allowed distance, otherwise forces a recall action from the host
+                if (!host.Transform.Coordinates.InRange(EntityManager, guardian.Transform.Coordinates, component.DistanceAllowed))
+                {
+                    OnGuardianManifestAction(uid, host.Uid);
+                }
+            }
+        }
```

Comment:
As above, also copy-paste is icky, just have it call into the same method.

Reply:
How?
I would gladly, but I couldn't pull it off.
You mean I should make a seperate recalling method and have the event trigger it?